src/3d/qgsglobechunkedentity.cpp

```diff
@@ -0,0 +1,406 @@
+/***************************************************************************
+  qgsglobechunkedentity.cpp
+  --------------------------------------
+  Date                 : March 2025
+  Copyright            : (C) 2025 by Martin Dobias
+  Email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsglobechunkedentity.h"
+
+#include <QByteArray>
+#include <QTimer>
+#include <QImage>
+#include <QtMath>
+
+#include <Qt3DCore/QEntity>
+#include <Qt3DRender/QAttribute>
+#include <Qt3DRender/QBuffer>
+#include <Qt3DRender/QGeometry>
+#include <Qt3DRender/QGeometryRenderer>
+#include <Qt3DRender/QTexture>
+#include <Qt3DRender/QTextureImage>
+#include <Qt3DExtras/QTextureMaterial>
+
+#include "qgs3dmapsettings.h"
+#include "qgschunkloader.h"
+#include "qgscoordinatereferencesystem.h"
+#include "qgscoordinatetransform.h"
+#include "qgsdistancearea.h"
+#include "qgsgeotransform.h"
+#include "qgsterraintextureimage_p.h"
+#include "qgsterraintexturegenerator_p.h"
+
+///@cond PRIVATE
+
+using namespace Qt3DCore;
+using namespace Qt3DRender;
+
+
+QEntity *makeGlobeMesh( double lonMin, double lonMax, double latMin, double latMax, int lonSliceCount, int latSliceCount, QImage textureQImage, QString textureDebugText )
+{
+  double lonRange = lonMax - lonMin;
+  double latRange = latMax - latMin;
+  double lonStep = lonRange / ( double ) ( lonSliceCount - 1 );
+  double latStep = latRange / ( double ) ( latSliceCount - 1 );
+
+  QgsCoordinateTransform ct( QgsCoordinateReferenceSystem( "EPSG:4326" ), QgsCoordinateReferenceSystem( "EPSG:4978" ), QgsCoordinateTransformContext() );
```

Comment:
Shouldn't this be using the inverse transform of `mGlobeCrsToLatLon` from QgsGlobeChunkLoaderFactory?